chainer/dataset/iterator.py

```diff
@@ -0,0 +1,72 @@
+class Iterator(object):
+
+    """Base class of all dataset iterators.
+
+    Iterator iterates over the dataset, yielding a minibatch at each
+    iteration. Minibatch is a list of examples. Each implementation should
+    implement an iterator protocol (e.g., the :meth:`__next__` method).
+
+    Note that, even if the iterator supports setting the batch size, it does
+    not guarantee that each batch always contains the same number of examples.
+    For example, if you let the iterator to stop at the end of the sweep, the
+    last batch may contain a fewer number of examples.
+
+    The interface between the iterator and the underlying dataset is not fixed,
+    and up to the implementation.
+
+    Each implementation should provide the following attributes (or possibly
+    read-only properties).
```

Comment:
Is the it a requirement that properties are read-only? I mean, what happens when the properties listed here are writable?